### PR TITLE
V5.0.0 release candidate

### PR DESCRIPTION
**Pull request by kevinkowa**
generated with 3.17.0 generator and sdk-major-release-2020-rc01 api def

### CODE REVIEW

File: src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs [hunk 1]
```diff
@@ -18,17 +18,18 @@
...
+using System;
 using IBM.Watson.VisualRecognition.v4.Model;
 
 namespace IBM.Watson.VisualRecognition.v4
 {
     public partial interface IVisualRecognitionService
     {
         DetailedResponse<AnalyzeResponse> Analyze(List<string> collectionIds, List<string> features, List<FileWithMetadata> imagesFile = null, List<string> imageUrl = null, float? threshold = null);
-        DetailedResponse<Collection> CreateCollection(string name = null, string description = null);
+        DetailedResponse<Collection> CreateCollection(string name = null, string description = null, TrainingStatus trainingStatus = null);
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs:29]: Lets remove this. should be readonly

**Review by mediumTaj — CHANGES_REQUESTED** [review]

File: MIGRATION-V5.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+* `newContext` type was changed from `Dictionary<string, object>` to `DialogNodeContext` in the `UpdateDialogNode()` method
+* `BulkClassify` method was addded
+
+##### Models Added
+
+`BulkClassifyOutput`, 
+`BulkClassifyResponse`, 
+`BulkClassifyUtterance`, 
+`DialogNodeContext`, 
+`DialogNodeOutputConnectToAgentTransferInfo`, `DialogNodeOutputGenericDialogNodeOutputResponseTypeConnectToAgent`, `DialogNodeOutputGenericDialogNodeOutputResponseTypeImage`, `DialogNodeOutputGenericDialogNodeOutputResponseTypeOption`, `DialogNodeOutputGenericDialogNodeOutputResponseTypePause`, `DialogNodeOutputGenericDialogNodeOutputResponseTypeSearchSkill`, `DialogNodeOutputGenericDialogNodeOutputResponseTypeText`, `RuntimeResponseGenericRuntimeResponseTypeConnectToAgent`,
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: fixed!

File: MIGRATION-V5.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+* Changed `toneInput` parameter type from `ToneInput` to `System.IO.MemoryStream` in `Tone()` method
+
+#### Visual Recognition v3
+
+No changes
+
+#### Visual Recognition v4
+
+* Added `trainingStatus` parameter to `CreateCollection()` method
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: Got it!

File: src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs [hunk 4]
```diff
@@ -18,17 +18,18 @@
...
         DetailedResponse<CollectionsList> ListCollections();
         DetailedResponse<Collection> GetCollection(string collectionId);
-        DetailedResponse<Collection> UpdateCollection(string collectionId, string name = null, string description = null);
+        DetailedResponse<Collection> UpdateCollection(string collectionId, string name = null, string description = null, TrainingStatus trainingStatus = null);
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs]: Fixed!

File: src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs (hunk 1 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/IVisualRecognitionService.cs]: Fixed!

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 5]
```diff
@@ -24,11 +24,101 @@
         <DebugSymbols>true</DebugSymbols>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Examples\**" />
+      <EmbeddedResource Remove="Examples\**" />
+      <None Remove="Examples\**" />
+    </ItemGroup>
```
Comment (mediumTaj) [inline src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:31]: should add tests
```
    <ItemGroup>
      <Compile Remove="Examples\**" />
      <Compile Remove="Test\**" />
      <EmbeddedResource Remove="Examples\**" />
      <EmbeddedResource Remove="Test\**" />
      <None Remove="Examples\**" />
      <None Remove="Test\**" />
    </ItemGroup>
```

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj [hunk 6]
```diff
@@ -24,11 +24,100 @@
         <DebugSymbols>true</DebugSymbols>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Examples\**" />
+      <EmbeddedResource Remove="Examples\**" />
+      <None Remove="Examples\**" />
+    </ItemGroup>
+
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj:32]: Should add tests
```
    <ItemGroup>
      <Compile Remove="Examples\**" />
      <Compile Remove="Test\**" />
      <EmbeddedResource Remove="Examples\**" />
      <EmbeddedResource Remove="Test\**" />
      <None Remove="Examples\**" />
      <None Remove="Test\**" />
    </ItemGroup>
```

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj [hunk 7]
```diff
@@ -24,11 +24,90 @@
         <DebugSymbols>true</DebugSymbols>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Examples\**" />
+      <EmbeddedResource Remove="Examples\**" />
+      <None Remove="Examples\**" />
+    </ItemGroup>
+
```
Comment (mediumTaj) [inline src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj:32]: Should add tests
```
    <ItemGroup>
      <Compile Remove="Examples\**" />
      <Compile Remove="Test\**" />
      <EmbeddedResource Remove="Examples\**" />
      <EmbeddedResource Remove="Test\**" />
      <None Remove="Examples\**" />
      <None Remove="Test\**" />
    </ItemGroup>
```

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 8]
```diff
@@ -24,11 +24,101 @@
...
     <ItemGroup>
         <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.1.0" />
-        <PackageReference Include="IBM.Watson.Common" Version="4.6.0" />
-        <PackageReference Include="JsonSubTypes" Version="1.6.0" />
+        <PackageReference Include="IBM.Watson.Common" Version="5.0.0" />
         <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+        <PackageReference Include="JsonSubTypes" Version="1.6.0" />
     </ItemGroup>
 
+    <ItemGroup>
```
Comment (mediumTaj) [inline src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:40]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 9]
```diff
@@ -24,11 +24,101 @@
...
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.IntegrationTests.Program.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.IntegrationTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.Program.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.IntegrationTests.Program.cs" />
+    </ItemGroup>
+    <ItemGroup>
```
Comment (mediumTaj) [inline src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:57]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj [hunk 10]
```diff
@@ -24,11 +24,90 @@
...
     <ItemGroup>
         <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.1.0" />
-        <PackageReference Include="IBM.Watson.Common" Version="4.6.0" />
+        <PackageReference Include="IBM.Watson.Common" Version="5.0.0" />
         <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
         <PackageReference Include="JsonSubTypes" Version="1.6.0" />
     </ItemGroup>
 
+    <ItemGroup>
+      <Compile Remove="Test\Unit\SpeechToTextServiceGeneratedUnitTests.cs" />
```
Comment (mediumTaj) [inline src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj:41]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj [hunk 11]
```diff
@@ -24,11 +24,90 @@
...
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.SpeechToText.v1.UnitTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Integration\SpeechToTextServiceIntegrationTest.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.SpeechToText.v1.IntegrationTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.SpeechToText.v1.IntegrationTests.Program.cs" />
+      <Compile Remove="Test\Integration\Properties\AssemblyInfo.cs" />
+    </ItemGroup>
+    <ItemGroup>
+      <None Remove="Test\.DS_Store" />
```
Comment (mediumTaj) [inline src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj:52]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj [hunk 12]
```diff
@@ -24,11 +24,100 @@
...
     <ItemGroup>
         <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.1.0" />
-        <PackageReference Include="IBM.Watson.Common" Version="4.6.0" />
-        <PackageReference Include="JsonSubTypes" Version="1.6.0" />
+        <PackageReference Include="IBM.Watson.Common" Version="5.0.0" />
         <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+        <PackageReference Include="JsonSubTypes" Version="1.6.0" />
     </ItemGroup>
 
+    <ItemGroup>
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj:40]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj [hunk 13]
```diff
@@ -24,11 +24,100 @@
...
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.VisualRecognition.v4.UnitTests.Program.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.VisualRecognition.v4.UnitTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Integration\VisualRecognitionServiceIntegrationTests.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.VisualRecognition.v4.IntegrationTests.AssemblyInfo.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\.NETCoreApp,Version=v2.0.AssemblyAttributes.cs" />
+      <Compile Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.VisualRecognition.v4.IntegrationTests.Program.cs" />
+      <Compile Remove="Test\Integration\Properties\AssemblyInfo.cs" />
+    </ItemGroup>
+    <ItemGroup>
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj:52]: I think we can remove this itemgroup if we add tests above

File: src/IBM.Watson.VisualRecognition.v4/Model/Collection.cs [hunk 14]
```diff
@@ -59,7 +59,7 @@ public class Collection
         /// Training status information for the collection.
         /// </summary>
         [JsonProperty("training_status", NullValueHandling = NullValueHandling.Ignore)]
-        public virtual TrainingStatus TrainingStatus { get; private set; }
+        public CollectionTrainingStatus TrainingStatus { get; set; }
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/Model/Collection.cs:62]: I think training status should stay as is - it was virtual bc it was readonly

**Review by mediumTaj — CHANGES_REQUESTED** [review]
A few more changes kevin

File: src/IBM.Watson.VisualRecognition.v4/Model/Collection.cs (hunk 14 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/Model/Collection.cs]: Got it!

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj (hunk 13 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj]: Fixed!

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj (hunk 12 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj]: Fixed!

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj (hunk 11 above)
Comment (kevinkowa) [reply src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj]: Fixed!

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj (hunk 10 above)
Comment (kevinkowa) [reply src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj]: Fixed!

File: src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj (hunk 6 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/IBM.Watson.VisualRecognition.v4.csproj:32]: Fixed!

File: src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj (hunk 7 above)
Comment (kevinkowa) [reply src/IBM.Watson.SpeechToText.v1/IBM.Watson.SpeechToText.v1.csproj:32]: Fixed!

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj (hunk 8 above)
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:43]: Fixed!

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj (hunk 9 above)
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:60]: Fixed!

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj (hunk 5 above)
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj:34]: Fixed!

File: MIGRATION-V5.md [hunk 15]
```diff
@@ -0,0 +1,250 @@
...
+
+##### Models Removed
+
+`VoiceModel`, 
+`VoiceModels`
+
+##### Model Properties Changed
+
+`Voice`
+* Change return type of `Customization()` from `VoiceModel` to `CustomModel`
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: It's a property, fixed!

File: MIGRATION-V5.md [hunk 16]
```diff
@@ -0,0 +1,250 @@
...
+* `includeCount` is now a parameter of the `ListIntents()` method
+* `includeCount` is now a parameter of the `ListExamples()` method
+* `includeCount` is now a parameter of the `ListCounterexamples()` method
+* `includeCount` is now a parameter of the `ListEntities()` method
+* `includeCount` is now a parameter of the `ListValues()` method
+* `includeCount` is now a parameter of the `ListSynonyms()` method
+* `includeCount` is now a parameter of the `ListDialogNodes()` method
+* `context` type was changed from `Dictionary<string, object>` to `DialogNodeContext` in the `CreateDialogNode()` method
+* `newContext` type was changed from `Dictionary<string, object>` to `DialogNodeContext` in the `UpdateDialogNode()` method
+* `BulkClassify` method was addded
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: Fixed!

File: MIGRATION-V5.md [hunk 17]
```diff
@@ -0,0 +1,250 @@
...
+
+`DialogNodeOutputGeneric`, `RuntimeResponseGeneric`
+* Added `AgentAvailable`, `AgentUnavailable`, and `TransferInfo` properties
+
+`DialogSuggestion`
+* `output` property type changed from `DialogSuggestionOutput` to `Dictionary<string, object>`
+
+#### Assistant v2
+
+* `BulkClassify` method was addded
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: Fixed!

File: MIGRATION-V5.md [hunk 18]
```diff
@@ -0,0 +1,250 @@
...
+`NluEnrichmentCategories`
+
+##### Model Properties Changed
+
+`NluEnrichmentFeatures`
+* Changed `Categories` property type from `NluEnrichmentCategories` to `Dictionary<string, object>`
+
+#### Discovery v2
+
+* `analyzeDocument()` method was addded
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply MIGRATION-V5.md]: Fixed!

**Pull request closed without merging**